packages/sdk-react/src/components/receivables/CreditNotesTable/CreditNotesTable.tsx

```diff
@@ -76,6 +77,29 @@ const CreditNotesTableBase = ({ onRowClick }: CreditNotesTableProps) => {
     setPaginationToken(undefined);
   };
 
+  // Adapted from https://mui.com/x/react-data-grid/column-dimensions/#autosizing-asynchronously
+  // setTimeout and flushSync are necessary for call order control
+  // Docs say:
+  // The Data Grid can only autosize based on the currently rendered cells.
+  // DOM access is required to accurately calculate dimensions
+  const gridApiRef = useGridApiRef();
+  useEffect(() => {
+    setTimeout(() => {
+      ReactDOM.flushSync(() => {
```

Comment:
Consider putting this logic in a separate hook, as we have some duplication here:

```ts
const useAutoSizeColumns = (gridApiRef, dependencies = []) => {
  useEffect(() => {
    const handleAutoSizeColumns = () => {
      setTimeout(() => {
        ReactDOM.flushSync(() => {
          setTimeout(() => {
            if (gridApiRef.current?.autosizeColumns) {
              gridApiRef.current.autosizeColumns({
                columns: ['amount'],
                includeHeaders: true,
                includeOutliers: true,
              });
            }
          }, 1);
        });
      }, 1);
    };

    handleAutoSizeColumns();
  }, [gridApiRef, ...dependencies]);
};
```

Reply:
Good point! Thanks! I've just pushed a code refactoring.